imbi/automations/sentry.py

```diff
@@ -0,0 +1,92 @@
+                    self.imbi_project_id)
+            else:
+                self.project = project
+        return self.errors
+
+    async def run(self) -> sentry.ProjectInfo:
+        project = await self.client.create_project(
+            self.project.namespace.sentry_team_slug, self.project.name)
+        try:
+            await self.db.execute(
+                'UPDATE v1.projects'
+                '   SET sentry_project_slug = %(slug)s,'
+                '       last_modified_at = CURRENT_TIMESTAMP,'
+                '       last_modified_by = %(username)s'
+                ' WHERE id = %(imbi_project_id)s', {
+                    'imbi_project_id': self.imbi_project_id,
+                    'slug': project.slug,
+                    'username': self.user.username,
+                }
+            )
+            if self.settings.get('project_link_type_id'):
+                await self.db.execute(
+                    '  INSERT INTO v1.project_links(project_id, link_type_id,'
+                    '                               created_by, url)'
+                    '       VALUES (%(imbi_project_id)s,'
+                    '               %(project_link_type_id)s, %(username)s,'
+                    '               %(url)s) '
+                    '  ON CONFLICT '
+                    'ON CONSTRAINT project_links_pkey'
+                    '    DO UPDATE'
+                    '          SET last_modified_at = CURRENT_TIMESTAMP,'
+                    '              last_modified_by = %(username)s,'
+                    '              url = %(url)s', {
+                        'imbi_project_id': self.imbi_project_id,
+                        'project_link_type_id':
+                            self.settings['project_link_type_id'],
+                        'url': project.link,
+                        'username': self.user.username,
+                    }
+                )
+            for name, value in project.keys.items():
+                await self.db.execute(
+                    '  INSERT INTO v1.project_secrets(project_id, name, value,'
+                    '                                 created_by)'
+                    '       VALUES (%(imbi_project_id)s, %(key_name)s,'
+                    '               %(key_value)s, %(username)s)'
+                    '  ON CONFLICT '
+                    'ON CONSTRAINT project_secrets_pkey'
+                    '    DO UPDATE'
+                    '          SET value = %(key_value)s,'
+                    '              last_modified_at = CURRENT_TIMESTAMP,'
+                    '              last_modified_by = %(username)s', {
+                        'imbi_project_id': self.imbi_project_id,
+                        'key_name': f'sentry_{name.lower()}',
+                        'key_value': self.application.encrypt_value(value),
+                        'username': self.user.username,
+                    }
+                )
+
+        except Exception:
```

Comment:
Why the bare Exception here?

Reply:
It is a little dirty ... _didn't realize that I left it in_.  The intent is to ensure that we remove the sentry project that we created if anything went awry when saving the information to the database.  Lemme look into tightening that up